PSEUDO API 1.md

```diff
@@ -0,0 +1,217 @@
+		Promise<sequence <Reality>> getRealities();
+		Promise<boolean> requestRealityChange(Reality reality); // resolves true if the request is accepted
+
+		Promise<XRFrameOfReference> requestFrameOfReference(XRFrameOfReferenceType type);
+
+		long requestFrame(XRFrameRequestCallback callback);
+		void cancelFrame(long handle);
+
+		Promise<void> endSession();
+
+		attribute EventHandler onblur;
+		attribute EventHandler onfocus;
+		attribute EventHandler onresetpose;
+		attribute EventHandler onended;
+		attribute EventHandler onrealitychanged;
+	};
+
+## Reality
+
+	interface Reality : EventTarget {
+		readonly atttribute DOMString realityName;
+		readonly attribute XRCoordinates stageLocation;
+		readonly attribute isPassthrough; // True if the Reality is a view of the outside world, not a fully VR
+
+		Promise<boolean> requestStageLocation(XRCoordinates coordinates);
+		Promise<boolean> requestResetStageLocation();
+
+		attribute EventHandler onchange;
+	};
+
+### Todo
+
+- configuration (e.g. change white balance on camera input, change options on map view)
+- offer a manifold aor a point cloud?
+
+
+## XRPointCloud
+
+	interface XRPointCloud {
+		readonly attribute Float32Array points;
+	}
+
+## XRLightEstimate
+
+	interface XRLightEstimate {
+		readonly attribute float ambientIntensity;
+		readonly attribute float ambientColorTemperature;
+	}
+
+## XRAnchor
+
+	interface XRAnchor {
+		readonly attribute long id;
+		readonly attribute Float32Array center; // x, y, z
+	}
+
+## XRPlaneAnchor
+
+	interface XRPlaneAnchor : XRAnchor {
+		readonly attribute Float32Array orientation; // quaternion?
+		readonly attribute Float32Array extent; // width, length
+	}
+
+## XRPresentationFrame
+
+	interface XRPresentationFrame {
+		readonly attribute FrozenArray<XRView> views;
+
+		readonly attribute boolean hasPointCloud;
+		readonly attribute XRPointCloud pointCloud;
+
+		readonly attribute boolean hasManifold;
+		readonly attribute XRManifold manifold;
+
+		readonly attribute boolean hasLightEstimate;
+		readonly attribute XRLightEstimate lightEstimate;
+		
+		long addAnchor(XRAnchor anchor);
+		void removeAnchor(long id);
+		XRAnchor? getAnchor(long id);
+		<sequence <XRAnchor>> getAnchors();
+
+		XRDisplayPose? getDisplayPose(XRCoordinateSystem coordinateSystem);
+	};
+
+
+- access camera image buffer aor texture
+
+## XRView
+
+	interface XRView {
+		readonly attribute XREye eye;
+		readonly attribute Float32Array projectionMatrix;
+
+		XRViewport? getViewport(XRLayer layer);
+	};
+
+## XRViewport
+
+	interface XRViewport {
+		readonly attribute long x;
+		readonly attribute long y;
+		readonly attribute long width;
+		readonly attribute long height;
+	};
+
+## XRCartographicCoordinates
+
+	interface XRCartographicCoordinates {
+		attribute float latitude;
+		attribute float longitude;
+		attribute float altitude;
+	}
+
+## XRCoordinateSystem
+
+	interface XRCoordinateSystem {
+		readonly attribute XRCartographicCoordinates? mapLocation;
```

Comment:
Also, I think we may want to consider saving some representation of error for location, as GPS and location APIs provide: it matters and needs to be exposed to developers can decide how they want to rely on location